services/viva-ms/lambdas/syncExpiryTime.js

```diff
@@ -1,123 +1,104 @@
 /* eslint-disable no-console */
 import to from 'await-to-js';
-import * as dynamoDb from '../../../libs/dynamoDb';
 
 import config from '../../../config';
-import params from '../../../libs/params';
 
+import * as dynamoDb from '../../../libs/dynamoDb';
+import log from '../../../libs/logs';
+import { getItem as getStoredUserCase } from '../../../libs/queries';
+import { statusTypes } from '../../../libs/caseStatuses';
 import { getFutureTimestamp, millisecondsToSeconds } from '../../../libs/timestampHelper';
 import {
-  DELETE_VIVA_CASE_AFTER_6_MONTH,
-  DELETE_VIVA_CASE_AFTER_45_DAYS,
-  DELETE_VIVA_CASE_AFTER_72_HOURS,
+  DELETE_VIVA_CASE_AFTER_6_MONTH as AFTER_6_MONTH,
+  DELETE_VIVA_CASE_AFTER_45_DAYS as AFTER_45_DAYS,
+  DELETE_VIVA_CASE_AFTER_72_HOURS as AFTER_3_DAYS,
 } from '../../../libs/constants';
 
-import log from '../../../libs/logs';
-
-const VIVA_CASE_SSM_PARAMS = params.read(config.cases.providers.viva.envsKeyName);
-
 export async function main(event, context) {
   const { caseKeys } = event.detail;
```

Comment:
We should add safe check here so that contents in detail is not undefined.

```JS
function noSuchProperty(key, object) {
  // test if a property doesn't exist in the current object
  return !object.hasOwnProperty(key)
}

if (noSuchProperty('caseKeys', event.detail)){
  // do something
}

const {caseKeys} = event.detail
 if(noSuchProperty('PK', caseKeys) || noSuchProperty('SK', caseKeys)) {
   // do something
 }

```

Reply:
Will be fixed in future task